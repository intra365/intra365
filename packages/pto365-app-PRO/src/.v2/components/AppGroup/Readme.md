

Sample

```js
<AppGroup  apps={["word","excel","unknown","teams"]} />
```