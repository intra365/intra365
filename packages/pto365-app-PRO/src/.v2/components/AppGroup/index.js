import React, { Component } from 'react';
import PropTypes from 'prop-types'
import { CommandBar } from 'office-ui-fabric-react/lib/CommandBar';
import { BaseComponent, memoizeFunction } from 'office-ui-fabric-react/lib/Utilities';
import { mergeStyles } from 'office-ui-fabric-react/lib/Styling';
import { ResizeGroup } from 'office-ui-fabric-react/lib/ResizeGroup';

import Utiltity from '../../utilities';
import TestData from '../../data/TestData';
import AppIconMini from '../AppIconMini'
import { IconButton } from 'office-ui-fabric-react/lib/Button';
import { CommandBarButton } from 'office-ui-fabric-react/lib/Button';
import Data from '../_Data'

let leftRightBoxClassName = mergeStyles({
  display: 'flex',
  justifyContent: 'space-between',
  whiteSpace: 'nowrap'
}) 

 let getNumberedBoxClassName = memoizeFunction((backgroundColor) => {
  return mergeStyles({
    display: 'inline-block',
    textAlign: 'center',
    textHeight:'10px',
    verticalAlign:'top',
    fontSize: '10px',
    lineHeight: '28px',

    height: '28px',
    width: '28px',
    marginLeft: '4px',
    marginRight: '4px',
    backgroundColor
  }) 
});
let BoxWithLabel = (props) => (
  <div className={getNumberedBoxClassName(props.backgroundColor)} title={props.label}><div style={{overflow:"clip"}}>{props.label.substring(0,2)}</div></div>
);

let renderApps = (apps) => {
  const result = [];
  var data = new Data()
  for (let i = 0; i < apps.length; i += 1) {

    var appData = data.getApp(apps[i])
    if (appData){
      result.push(<div className={getNumberedBoxClassName(appData.color)} key={i}>
      <AppIconMini backgroundColor={appData.color} name={appData.name} iconUrl={appData.icon}  />
      </div>);
    }else
    {
   //result.push(<BoxWithLabel backgroundColor="#888888" label={apps[i]} key={i}/>)
    }
  }
  return result;
}


let LeftRightBoxSet = (props) => (

    <div>{renderApps(props.apps)}</div>
)
/**
 * Handle presentation of a group of icons 
 *
 * @export
 * @class AppGroup
 * @extends {Component}
 */
export default class AppGroup extends Component { 

    static propTypes  = {
        about : PropTypes.string ,  // sample - remove or rename
        apps : PropTypes.arrayOf(PropTypes.string).isRequired
    }
    

    constructor(props) {
        super(props);
    }
     
    
    

     onReduceData(props)  {
      if (props.leftCount === 0 && props.rightCount === 0) {
        return undefined
      }
    
      let result
      if (props.leftCount > props.rightCount) {
        result = { ...props, leftCount: props.leftCount - 1 };
      } else {
        result = { ...props, rightCount: props.rightCount - 1 };
      }
    
      // Update the cache key
      return { ...result, cacheKey: `${result.leftCount + result.rightCount}` };
    }
    

    copyArray = (array) => {
      const newArray = [];
      for (let i = 0; i < array.length; i++) {
        newArray[i] = array[i];
      }
      return newArray;
    }
    /**
     * Required method return the output of the component
     *
     * @returns
     * @memberof AppGroup
     */
    render() {


      if(!this.props.apps) return null
        const customButton = (props) => {
            const buttonOnMouseEnter = () => console.log(`${props.text} hovered`);
            return (<div>
              <CommandBarButton
                onMouseEnter={buttonOnMouseEnter}
                {...props}
                styles={{
                  ...props.styles,
                  icon: { color: 'red' }
                }}
              /></div>
            );
          };    
        var tasks = Utiltity.GroupTasksBySubject(TestData.tasks)
        var items = Utiltity.EnsureNameAndApplyKeyValue(tasks,"subject")
        
        const data  = { apps: this.copyArray(this.props.apps),leftCount: 10, rightCount: 0, cacheKey: '10' }
        return (

         
        
            <ResizeGroup
              data={data}
              
              onRenderData={(scaledData) => <LeftRightBoxSet {...scaledData} />}
              onReduceData={this.onReduceData}
            />
        
        
 
        )
    }
}

