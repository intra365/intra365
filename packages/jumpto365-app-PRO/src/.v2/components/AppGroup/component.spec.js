import React, { Component } from 'react';

import AppGroup from '.'
var apps = ["word","excel","teams","unknown"]
test('Check that it renders', () => {

    const wrapper = shallow(<AppGroup apps={apps}  about="🍕"/>);

    expect(wrapper).toMatchSnapshot();
 
}

)





test('react on click', () => {
    const spy = sinon.spy();
    const wrapper = mount(<AppGroup apps={apps} onClick={spy} />);
    wrapper
    .find("div")
    .first()
    .simulate("click");
    
    expect(spy.calledOnce).toBe(false);
 
}

) 
