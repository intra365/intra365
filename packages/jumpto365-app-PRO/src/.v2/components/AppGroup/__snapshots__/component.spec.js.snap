// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppGroup
    about="🍕"
    apps={
        Array [
            "word",
            "excel",
            "teams",
            "unknown",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "data": Object {
        "apps": Array [
          "word",
          "excel",
          "teams",
          "unknown",
        ],
        "cacheKey": "10",
        "leftCount": 10,
        "rightCount": 0,
      },
      "onReduceData": [Function],
      "onRenderData": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {
          "apps": Array [
            "word",
            "excel",
            "teams",
            "unknown",
          ],
          "cacheKey": "10",
          "leftCount": 10,
          "rightCount": 0,
        },
        "onReduceData": [Function],
        "onRenderData": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
